libraries/standard/http/include/http_client.h

```diff
@@ -0,0 +1,357 @@
+     */
+    uint8_t* pBuffer;
+    size_t bufferLen; /**< The length of pBuffer in bytes. */
+    
+    /**
+     * @brief The actual amount of headers in the buffer, in bytes. This field
+     * is updated by the HTTP Client library functions #HTTPClient_AddHeader,
+     * and #HTTPClient_InitializeRequestHeaders.
+     */
+    size_t headersLen;
+} HTTPRequestHeaders_t;
+
+/**
+ * @brief Configurations of the initial request headers.
+ */
+typedef struct HTTPRequestInfo
+{
+    /**
+     * @brief The HTTP request method i.e. "GET", "POST", "PUT", or "HEAD". 
+     */
+    const char* method;
+    size_t methodLen; /**< The length of the method in bytes. */
+    
+    /**
+     * @brief The Request-URI to the objects of interest, i.e. "/path/to/item.txt".
+     */
+    const char* pPath;
+    size_t pathLen; /**< The length of the path in bytes. */
+
+    /**
+     * @brief The DNS resolvable name of the host or server, i.e. 
+     * "s3.amazonaws.com"
+     * 
+     * The host does not have a "https://" or "http://" prepending.
+     */
+    const char* pHost;
+    size_t hostLen; /**< The length of the host in bytes. */
+   
+    /**
+     * @brief Flags to activate other request header configurations. 
+     */
+    uint32_t flags;
+} HTTPRequestInfo_t;
+
+
+
+/**
+ * @brief Callback to intercept headers during the first parse through of the 
+ * response as it is received from the network.
+ */
+typedef struct httpResponseParsingCallback {
+    /**
+     * @brief Invoked when both a header field and its associated header value are found.
+     * @param[in] pContext User context.
+     * @param[in] fieldLoc Location of the header field name in the response buffer.
+     * @param[in] fieldLen Length in bytes of the field name.
+     * @param[in] valueLoc Location of the header value in the response buffer.
+     * @param[in] valueLen Length in bytes of the value.
+     * @param[in] statusCode The HTTP response status-code.
+     */
+    void (*onHeaderCallback)( void* pContext, const char* fieldLoc, size_t fieldLen, const char* valueLoc, size_t valueLen, uint16_t statusCode );
+
+    /* Private context for the application. */
+    void *pContext;
+} HTTPClient_HeaderParsingCallback_t;
+
+/**
+ * @brief Represents an HTTP response.
+ */
+typedef struct HTTPResponse
+{
+    /**
+     * @brief Buffer for both the raw HTTP header and body. 
+     * 
+     * This buffer is supplied by the application.
+     * 
+     * This buffer is owned by the library during  #HTTPClient_Send. This buffer
+     * should not be modifed until after this functions return.
+     * 
+     * For optimization this buffer may be re-used with the request. The user 
+     * can re-use this buffer for the storing the request headers in 
+     * #HTTPRequestHeaders_t.pBuffer.
+     */
+    uint8_t* pBuffer;
+    size_t bufferLen; /**< The length of the response buffer in bytes. */
+
+    /**
+     * @brief Optional callback for intercepting the header during the first 
```

Comment:
I'd add "Set to NULL to disable." at the end of the description.

Reply:
Done.